### PR TITLE
Use external.sh and build_image.sh for more docker image tag coverage

### PR DESCRIPTION
**Pull request by smlambert**
This PR (replacing #1619) is an initial switch over for some external tests to use the new scripts which allow the use of any of the docker image tags that are produced and pushed to Docker hub.  

This is ultimately to facilitate the testing of the unofficial images so that we can create a repeatable QA process for verifying them and getting them pushed into the official Docker hub repo.

### CODE REVIEW

**Comment by smlambert** [comment]
Camel with ubi / jdk11:
Grinder #[2367](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/2367)

Derby with alpine / jdk11:
Grinder #[2368](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/2368)

Jenkins with centos / jdk11:
Grinder #[2369](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/2369)

Kafka with nightly / jdk8:
Grinder #[2370](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/2370)

**Comment by smlambert** [comment]
Things that are still missing:

- external.sh should take 2 report parameters (1 for src and 1 for dst info as not all apps use surefire-reports as src, and need to pass the REPORTDIR from TKG as dst)

- the run should actually fail if there is a failure during the build or execute stage and right now it is not

File: external/external.sh [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+if [ $command_type == "run" ]; then
+	docker run $docker_args adoptopenjdk-$test-test:${JDK_VERSION}-$package-$docker_os-${JDK_IMPL}-$build_type
+fi
+
+if [ $report != "false" ]; then
+	docker cp adoptopenjdk-$test-test:${JDK_VERSION}-$package-$docker_os-${JDK_IMPL}-$build_type:/testResults/surefire-reports $report/external_test_reports;
+fi
+
+if [ $command_type == "clean" ]; then
+	docker rmi -f adoptopenjdk-$test-test:${JDK_VERSION}-$package-$docker_os-${JDK_IMPL}-$build_type
```
Comment (sophia-guo) [inline external/external.sh]: Might be better to also include removing containers.

File: external/external.sh [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+if [ $command_type == "build" ]; then
+	source $(dirname "$0")/build_image.sh $test $version $impl $docker_os $package $build_type
+fi
+
+if [ $command_type == "run" ]; then
+	docker run $docker_args adoptopenjdk-$test-test:${JDK_VERSION}-$package-$docker_os-${JDK_IMPL}-$build_type
+fi
+
+if [ $report != "false" ]; then
```
Comment (sophia-guo) [inline external/external.sh]: Could we nest this report to if [ $command_type == "run" ] condition? Personally I think that's a bit more clearer.

Comment (smlambert) [reply external/external.sh]: yes, will do

File: external/external.sh (hunk 1 above)
Comment (james-crowley) [reply external/external.sh]: @sophia-guo Are you talking about removing the base containers? Like `adoptopenjdk:openjdk13-openj9`? Cause the above command will remove the test container built upon the `adoptopenjdk`, but not the base container itself.

**Comment by james-crowley** [comment]
> Jenkins with centos / jdk11:
Grinder #2369 - this tries to do the right thing, which is to build a centos based image, but that fails due to missing manifest, see console output:
07:49:55 [exec] manifest for adoptopenjdk/openjdk11:centos-nightly not found 07:49:55 [exec] Result: 1

@smlambert Do you have an issue open against https://github.com/AdoptOpenJDK/openjdk-docker/? I think we had this issue opened, https://github.com/AdoptOpenJDK/openjdk-docker/issues/268, but I am unsure what came of it.

**Comment by smlambert** [comment]
re: https://github.com/AdoptOpenJDK/openjdk-tests/pull/1660#issuecomment-593657452 I did not raise a new issue in openjdk-docker yet as the Jenkins server was unhealthy over the weekend, which may have caused a blip... was going to run Grinders again in the morning to see, then raise if still existed

File: external/external.sh (hunk 1 above)
Comment (sophia-guo) [reply external/external.sh]: Sorry, I didn't realize that we actually using --rm to delete anonymous test containers. That is being said all is good.

Comment (smlambert) [reply external/external.sh]: ok sweet

**Comment by smlambert** [comment]
I can update some of our test jobs tonight after a few more grinder runs before merging this, thanks for the review!

**Comment by smlambert** [comment]
Still hitting the same errors in some Grinders 
20:07:32       [exec] manifest for adoptopenjdk/openjdk11:centos-nightly not found
20:07:32       [exec] Result: 1
so have reopened https://github.com/AdoptOpenJDK/openjdk-docker/issues/281

File: external/external.sh [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+parseCommandLineArgs "$@"
+
+if [ $command_type == "build" ]; then
+	source $(dirname "$0")/build_image.sh $test $version $impl $docker_os $package $build_type
+fi
+
+if [ $command_type == "run" ]; then
+	docker run $docker_args adoptopenjdk-$test-test:${JDK_VERSION}-$package-$docker_os-${JDK_IMPL}-$build_type
+	if [ $report != "false" ]; then
+		docker cp adoptopenjdk-$test-test:${JDK_VERSION}-$package-$docker_os-${JDK_IMPL}-$build_type:/testResults/surefire-reports $report/external_test_reports;
```
Comment (sophia-guo) [inline external/external.sh]: Should be `docker cp container-name` not image name?

File: external/external.sh (hunk 1 above)
Comment (sophia-guo) [reply external/external.sh]: Currently there are two different scenarios. For tests do not need copy the results we can use --rm to automatically remove the container when it exits. For tests need to copy the test results we can't remove the container when exists. Instead we assign the container name when do `docker run --name`.  And  the container name will be used when doing `docker cp containername:directory....`.  In this scenario when doing clean the test container should also be removed. Sorry I just realized this.

Comment (smlambert) [reply external/external.sh]: ah that explains it.  Let me attempt to generalize this enough to make it possible.  Essentially I would like all test results from all apps made available... but I recognize some may not be in obvious locations

**Comment by smlambert** [comment]
Next grinders:

Camel with ubi / jdk11:
Grinder [#2438](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/2438)

Derby with alpine / jdk11:
Grinder [#2439](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/2439)

Jenkins with centos / jdk11:
Grinder [#2440](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/2440)

Kafka with nightly / jdk8:
Grinder [#2441](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/2441)

**Comment by smlambert** [comment]
I will create a follow-up PR for proper handling of reports (in which we may also want to include cores, which in some cases go to a different dir than test results, so will need special handling if we want them).

**Pull request merged**